### PR TITLE
Remove translog retention policy

### PR DESCRIPTION
**Pull request by dnhatn**
We no longer need to retain the extra translog for peer recovery as we have switched using Lucene index exclusively in 8.0. This change removes the translog retention policy.

I think we can also remove the async trim translog task. I will do it in a separate PR.

Relates #50775

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/IndexSettings.java [hunk 1]
```diff
@@ -248,8 +248,7 @@
      * This setting will be ignored if soft-deletes is used in peer recoveries (default in 7.4).
      **/
     public static final Setting<TimeValue> INDEX_TRANSLOG_RETENTION_AGE_SETTING =
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/index/IndexSettings.java]: should we deprecate these settings in 7 and disallow setting them on 8.x indices?

We should also add docs in this PR saying that these settings are no longer supported and are effectively ignored

File: server/src/main/java/org/elasticsearch/index/translog/TranslogDeletionPolicy.java [hunk 2]
```diff
@@ -156,66 +134,9 @@ private synchronized void releaseTranslogGen(long translogGen) {
...
-            totalFiles++;
-            minGen = readers.get(i).generation;
-        }
-        return minGen;
-    }
-
-    protected long currentTime() {
-        return System.currentTimeMillis();
+    synchronized long minTranslogGenRequired() {
+        return Math.min(getMinTranslogGenRequiredByLocks(), minTranslogGenerationForRecovery);
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/index/translog/TranslogDeletionPolicy.java]: 💯

File: server/src/test/java/org/elasticsearch/index/IndexServiceTests.java [hunk 3]
```diff
@@ -367,70 +361,6 @@ public void testRescheduleAsyncFsync() throws Exception {
         assertNotNull(indexService.getFsyncTask());
     }
 
-    public void testAsyncTranslogTrimActuallyWorks() throws Exception {
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/index/IndexServiceTests.java]: why remove this test (and the one below)? We have not removed this functionality and still need it, no?

**Review by ywelsch — COMMENTED** [review]
Great PR stats. I've left some minor comments.

File: server/src/main/java/org/elasticsearch/index/IndexSettings.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/index/IndexSettings.java]: > should we deprecate these settings in 7 and disallow setting them on 8.x indices? We should also add docs in this PR saying that these settings are no longer supported and are effectively ignored


Yup, It was planned in the next step :)

File: server/src/test/java/org/elasticsearch/index/IndexServiceTests.java (hunk 3 above)
Comment (dnhatn) [reply server/src/test/java/org/elasticsearch/index/IndexServiceTests.java]: restored in https://github.com/elastic/elasticsearch/pull/51417/commits/82255aae678c57e1b0ab4c67d7e21be7266cfb7f.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
run elasticsearch-ci/2

**Comment by dnhatn** [comment]
Thanks Yannick.

**Pull request merged**